### PR TITLE
Add deprecation notices, deprecation warning on install

### PR DESCRIPTION
**Pull request by awharn**
**What It Does**

Adds a deprecation notice to the readme, and on the post-install hooks.

**How to Test**

Install the IMS plug-in

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [ ] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 

Same question as with the `zowe-v1-lts` (publish as a `release-patch` and deprecate the package?)

File: src/pluginLifeCycle.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+* SPDX-License-Identifier: EPL-2.0                                                *
+*                                                                                 *
+* Copyright Contributors to the Zowe Project.                                     *
+*                                                                                 *
+*/
+
+import { AbstractPluginLifeCycle, Logger } from "@zowe/imperative";
+
+class PluginLifeCycle extends AbstractPluginLifeCycle {
+    public postInstall(): void | Promise<void> {
```
Comment (traeok) [inline src/pluginLifeCycle.ts:15]: not requesting any change, but the return types for `postInstall` and `preUninstall` can be reduced to just `void`

**Review by traeok — APPROVED** [review]
LGTM, thanks @awharn!

File: README.md
Comment (adam-wolfe) [inline README.md:1]: The message looks good, but could you also fix the occurrences of "rerequirements" in this readme?

**Review by gejohnston — APPROVED** [review]

**Pull request merged**